src/ui/public/test_harness/test_sharding/setup_test_sharding.js

```diff
@@ -0,0 +1,50 @@
+import { uniq, defaults } from 'lodash';
+
+import { findTestBundleUrl } from './find_test_bundle_url';
+import { getShardingParamsFromUrl } from './get_sharding_params_from_url';
+import { setupRootDescribeCallFilter } from './root_describe_call_filter';
+import { getShardNum } from './get_shard_num';
+
+const DEFAULT_PARAMS = {
+  shards: 1,
+  shard_num: 1
+};
+
+export function setupTestSharding() {
+  const pageUrl = window.location.href;
+  const bundleUrl = findTestBundleUrl();
+
+  // read config from the page url, and from the
+  // bundle url, so that we can override params
+  // in dev mode
+  const params = defaults(
+    {},
+    getShardingParamsFromUrl(pageUrl),
+    getShardingParamsFromUrl(bundleUrl),
+    DEFAULT_PARAMS
+  );
+
+  // map query string params to local vars
+  const { shards: shardTotal, shard_num: shardNum } = params;
+  if (shardNum < 1 || shardNum > shardTotal) {
+    throw new TypeError(`shard_num param of ${shardNum} must be greater 0 and less than the total, ${shardTotal}`);
+  }
+
+  // track and log the number of ignored describe calls
+  const ignoredDescribeShards = [];
+  before(() => {
+    const ignoredCount = ignoredDescribeShards.length;
+    const ignoredFrom = uniq(ignoredDescribeShards).join(', ');
+    console.log(`Ignored ${ignoredCount} root suites from ${ignoredFrom}`);
+  });
+
+  // Filter root level describe statements as they come
+  setupRootDescribeCallFilter(describeName => {
```

Comment:
@cjcenizal After we talked, I moved to this style. Putting the comments in this file made it less readable, so I pushed the walking/intercepting down to the lower module and just pulled out the filter bit.